packages/snapshot/src/manager.ts

```diff
@@ -23,7 +23,7 @@ export class SnapshotManager {
     addSnapshotResult(this.summary, result)
   }
 
-  resolvePath(testPath: string): string {
+  resolvePath(testPath: string, context?: any): string {
```

Comment:
Maybe we can use a generic here to allow a custom type? `<T>`?

Reply:
Technically we need to synchronize `SnapshotManager.resolvePath` and `SnapshotStateOptions.resolveSnapshotPath`, so if we want to go strict, `SnapshotManager<T>` and `SnapshotStateOptions<T>` but this looks too much. In my opinion, `any` looks fine here. Otherwise, we should do larger rework around this.

Reply:
I don't want to go more strict; I want to catch possible errors easily, so `resolvePath<T>` is enough for that I think

Reply:
Things are tangled up now and I don't see if using T is effective in current logic. This is an another take https://github.com/vitest-dev/vitest/pull/6800/commits/67fa38248f2bb48c5d9655d9761be9ae4b82c58b but doesn't it looks silly?

Reply:
Yeah, that might be the only saver. I pushed a final take https://github.com/vitest-dev/vitest/pull/6800/commits/55f6a396dacf781c3679779c37dde7e6860a8c36